lib/elixir/lib/calendar/time.ex

```diff
@@ -365,12 +365,20 @@ defmodule Time do
   end
 
   @doc """
-  Converts the `Time` struct to a different calendar.
+  Converts given `time` to a different calendar.
 
   Returns `{:ok, time}` if the conversion was successful,
   or `{:error, reason}` if it was not, for some reason.
+
+  ## Examples
+
+  Imagine someone implements `Calendar.Julian`:
+
+      iex> Time.convert(~T[13:30:15], Calendar.Julian)
+      {:ok, %Time{calendar: Calendar.Julian, hour: 13, minute: 30, second: 15, microsecond: {0, 0}}}
+
   """
-  @spec convert(Calendar.time, Calendar.calendar) :: {:ok, Time.t} | {:error, atom}
+  @spec convert(Calendar.time, Calendar.calendar) :: {:ok, t} | {:error, atom}
   def convert(%Time{calendar: calendar} = time, calendar) do
```

Comment:
We are matching on the struct, so we need to either change the implementation or change the type to `t`.